File: zx-sn-ping.js
const { spawn } = require('child_process');
const [,, subnetArg, durationStr] = process.argv;

if (!subnetArg || !durationStr) {
  console.log("Usage: node ZX-SN-PING.js <subnet/mask> <duration_in_seconds>");
  process.exit(1);
}

const cidrToIps = (cidr) => {
  const [base, bits] = cidr.split('/');
  const baseParts = base.split('.').map(Number);
  const maskBits = parseInt(bits, 10);
  const hosts = 2 ** (32 - maskBits);

  const baseNum =
    (baseParts[0] << 24) |
    (baseParts[1] << 16) |
    (baseParts[2] << 8) |
    (baseParts[3]);

  const ips = [];
  for (let i = 1; i < hosts - 1; i++) {
    const ip = baseNum + i;
    const ipStr = [
      (ip >> 24) & 255,
      (ip >> 16) & 255,
      (ip >> 8) & 255,
      ip & 255
    ].join('.');
    ips.push(ipStr);
  }

  return ips;
};

const allIps = cidrToIps(subnetArg);
const totalIps = allIps.length;
const duration = parseInt(durationStr); // in SECONDS
const startTime = Date.now();

let totalPings = 0;
let onlineIps = new Set();

function formatUptime(ms) {
  const s = Math.floor(ms / 1000);
  const h = Math.floor(s / 3600);
  const m = Math.floor((s % 3600) / 60);
  const secs = s % 60;
  return `${String(h).padStart(2, '0')}:${String(m).padStart(2, '0')}:${String(secs).padStart(2, '0')}`;
}

function pingIp(ip) {
  const ping = spawn('ping', ['-c', '1', '-s', '64', '-W', '1', ip]);

  let responded = false;

  ping.stdout.on('data', (data) => {
    if (data.toString().includes('bytes from')) {
      onlineIps.add(ip);
      responded = true;
    }
  });

  ping.on('close', () => {
    if (!responded) {
      onlineIps.delete(ip);
    }
    totalPings++;
  });
}

function updateDisplay() {
  process.stdout.write('\x1Bc'); // clear screen
  console.log(`ZX-SN-PING | T.ME/STSVKINGDOM`);
  console.log(`--------------------------------------------------`);
  console.log(`pings   : ${totalPings}`);
  console.log(`bytes   : 64`);
  console.log(`total ips : ${totalIps}`);
  console.log(`online  : ${onlineIps.size}/${totalIps}`);
  console.log(`offline : ${totalIps - onlineIps.size}`);
  console.log(`running : ${formatUptime(Date.now() - startTime)}`);
}

let elapsedSeconds = 0;

const tick = setInterval(() => {
  elapsedSeconds++;

  // ping all IPs every tick
  allIps.forEach(pingIp);

  updateDisplay();

  if (elapsedSeconds >= duration) {
    clearInterval(tick);
    updateDisplay();
    console.log(`\nZX-SN-PING complete.`);
    process.exit(0);
  }
}, 1000);

File: zx-sn-ping.py
import os
import sys
import time
import ipaddress
import subprocess
from datetime import datetime

if len(sys.argv) != 3:
    print("Usage: python3 ZX-SN-PING.py <subnet> <duration_in_seconds>")
    sys.exit(1)

subnet = sys.argv[1]
duration = int(sys.argv[2])

try:
    net = ipaddress.ip_network(subnet, strict=False)
except ValueError:
    print("Invalid subnet.")
    sys.exit(1)

ip_list = [str(ip) for ip in net.hosts()]
total_ips = len(ip_list)
start_time = time.time()

online = set()
offline = set()
total_pings = 0

def clear():
    os.system('clear' if os.name == 'posix' else 'cls')

def ping(ip):
    try:
        output = subprocess.check_output(
            ["ping", "-c", "1", "-W", "1", ip],
            stderr=subprocess.DEVNULL,
            universal_newlines=True
        )
        return "bytes from" in output
    except subprocess.CalledProcessError:
        return False

while time.time() - start_time < duration:
    current_online = set()
    for ip in ip_list:
        is_up = ping(ip)
        total_pings += 1
        if is_up:
            current_online.add(ip)
        else:
            offline.add(ip)

    online = current_online
    offline = set(ip_list) - online
    running_time = int(time.time() - start_time)

    clear()
    print("ZX-SN-PING | T.ME/STSVKINGDOM")
    print("--------------------------------------------------")
    print(f"pings:      {total_pings}")
    print(f"bytes:      64")
    print(f"total ips:  {total_ips}")
    print(f"online:     {len(online)}/{total_ips}")
    print(f"offline:    {len(offline)}")
    print(f"running:    {running_time} sec")
    print("--------------------------------------------------")
    print("TEAM S.T.S")
    print("t.me/fbigovv")
    print("t.me/stsgov")
    print("t.me/stsvkingdom")
    print("t.me/tspvkingdom")

    time.sleep(1)
